pkg/controllers/elasticsearch/nodepool/resources.go

```diff
@@ -133,6 +133,18 @@ func elasticsearchPodTemplateSpec(controllerName string, c *v1alpha1.Elasticsear
 	plugins := strings.Join(c.Spec.Plugins, ",")
 	nodePoolLabels := util.NodePoolLabels(c, np.Name, np.Roles...)
 
+	esImage := c.Spec.Image
+	if esImage.FsGroup == 0 &&
+		esImage.ImageSpec.PullPolicy == "" &&
+		esImage.ImageSpec.Repository == "" &&
+		esImage.ImageSpec.Tag == "" {
```

Comment:
* Why check for FsGroup here?
* Could you instead compare esImage.ImageSpec to the zero value of ImageSpec?
* What if I supply FsGroup, but omit the ImageSpec? It looks like the pod will fail to start.

Reply:
1) Changing the `version` attribute is expected use. This should cause navigator to begin an upgrade/downgrade/report an error that upgrade/downgrade isn't possible.

2) Nope - provide the `version` and everything else is handled for you. As soon as you specify either FsGroup, PullPolicy, Repository or Tag however, you've gone into manual mode. Manual mode *might* be removed at some point in the future (as discussed today)

3) As above, fsGroup is checked because if specified, you've switched to 'manual mode' (and so all the others are required too). Perhaps this will make a bit more sense once we get the validation PR (#199) merged.

4) Yep potentially we could just check the zero values - or as you suggested earlier, make this field a pointer.

5) Yep you are correct. FsGroup = manual mode. Remember users never see `imageSpec` as a field (as it's json tag is: `json:",inline"`)